<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="rviz"  default="true"/>

  <!-- Load Params According to Robot -->
  <rosparam file="$(find pick_and_place)/config/params_TIAGO.yaml" command="load" />

  <!-- Pick & place server -->
  <node name="pick_and_place_server" pkg="pick_and_place" type="pick_and_place_server.py" output="screen" />

  <!-- 2D Detector -->
  <include file="$(find object_detector)/launch/object_2D.launch">
    <arg name="SOURCE" value="/xtion/rgb/image_raw" />
    <arg name="DEPTH_INPUT" value="/xtion/depth_registered/image_raw" />
    <arg name="CAMERA_INFO" value="/xtion/rgb/camera_info" />
    <arg name="ROS_INPUT" value="True" />
    <arg name="USE_ACTIVE_FLAG" value="False" />
    <arg name="VERBOSE" value="False" />
    <arg name="DEPTH_ACTIVE" value="True" />
    <arg name="MODELS_PATH" default="$(find object_detector)/models/gazebo/" />
    <arg name="LABELS_PATH" default="$(find object_detector)/models/gazebo/label_map.pbtxt" />
    <arg name="CAMERA_FRAME" value="xtion_rgb_optical_frame" />
  </include>

  <!-- 3D Detector -->
  <include file="$(find object_detector)/launch/object_3D.launch">
    <arg name="MAP_FRAME" default="map" />
    <arg name="CAMERA_FRAME" default="xtion_rgb_optical_frame" />
    <arg name="map_to_odom"  default="false"/>
    <arg name="odom_to_internal_odom"  default="false"/>
    <arg name="POINT_CLOUD_TOPIC" default="/xtion/depth_registered/points" />
  </include>

  <!-- Manipulation Server -->
  <node name="manipulation_server" pkg="pick_and_place" type="manipulationServer.py" output="screen" />
  
  <!-- GPD -->
	<node name="detect_grasps_server_samples" pkg="gpd_ros" type="detect_grasps_server_samples" output="screen" respawn="true">
    <param name="config_file" value="/catkin_home/src/manipulation/pick_and_place/config/ros_eigen_params.cfg" />
    <param name="rviz_topic" value="plot_grasps" />
  </node>
  
  <!-- Manipulation Client | Running in separate shell -->
  <!-- <node name="manipulation_client" pkg="pick_and_place" type="manipulationClient.py" output="screen" /> -->
  <node name="rviz" pkg="rviz" type="rviz" if="$(arg rviz)"
          args="-d $(find pick_and_place)/config/all.rviz"/>


</launch>
